2.2.1 Release Notes Draft

Colin1841 opened the pull request:
## Jira Ticket

https://jira.d2iq.com/browse/D2IQ-88896

## Description of changes being made
For this issue: https://jira.d2iq.com/browse/COPS-7275
We ARE NOT adding this as a COPS fix, only adding Containerd to the Component Updates table.

**REMINDER!!** For 2.2 versions, we're doing a combined Release Notes with the same file posted to both Kommander and Konvoy. To alleviate confusion of working in duplicate pages, we'll work on this one page and then duplicate it before merging. 

### Preview

You can preview the docs build using the following URL, adding the PR # where specified:
http://docs-d2iq-com-pr-4475.s3-website-us-west-2.amazonaws.com/

faiq commented on pages/dkp/konvoy/2.2/release-notes/2.2.1/index.md:
```diff
@@ -0,0 +1,105 @@
+| ----------------------- | ---------------------------------------------- | ------- | ----------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Cert Manager            | cert-manager                                   | 1.7.1   | - chart: 1.7.1 `<br>`- cert-manager: 1.7.1                                                                                                          |
+| Chartmuseum             | chartmuseum                                    | 3.6.2   | - chart: 3.6.2 `<br>`- chartmuseum: 3.6.2                                                                                                           |
+| Containerd              | containerd                                     | 1.4.13  |                                                                                                                                                       |
```
I don't know if this is going to be bumped up in this release. cc @dkoshkin and @dlipovetsky

cgawron1 replied:
This component did not get updated in 2.2.1, Please omit this. Containerd is still on  1.14.11

faiq replied:
We are not planning on including a change to the containerd version for the 2.2.1 release

cmastr replied:
@Colin1841  while you were on vacation, we decided NOT to hold 2.2.1 for this. We will fix the issue with containerd in a future patch.

faiq commented on pages/dkp/konvoy/2.2/release-notes/2.2.1/index.md:
```diff
@@ -0,0 +1,105 @@
+### kube-oidc-proxy error: certificate signed by unknown authority (COPS-7217)
+
+When adding a new Attached Cluster to the Management Cluster, using a custom domain and TLS certificate issued by Let's Encrypt, the kube-oidc-proxy helm chart in the Attached Cluster did not complete installation and the associated pod returned an error.
+
```
cc @dkoshkin

Colin1841 replied:
@faiq - Is there an associated COPS for this?

faiq replied:
no cops for this we just backported this

cmastr replied:
We have to have a COPS ticket to include it in the documentation. Backport from where?

faiq replied:
https://jira.d2iq.com/browse/D2IQ-87527
https://github.com/mesosphere/konvoy-image-builder/pull/300
https://github.com/mesosphere/konvoy2/pull/1275

gvasquezvargas commented on pages/dkp/konvoy/2.2/release-notes/2.2.1/index.md:
```diff
@@ -0,0 +1,105 @@
+
+### Certs showing as updated but not reloading in Kommander pods (COPS-7212)
+
+Under certain conditions, when certificate objects from a bundle are not expired, the pods in the cluster were not reloading with new certificates and might show the current certificates as expired.
```
```suggestion
Some Kommander versions do not properly handle certificate renewal for the Cluster CA and certificates that are created for Kommander applications. When this occurs, some Kommander applications and pods fail to receive the renewed certificate information, causing them to stop working upon expiration. This problem has been corrected.
```

Colin1841 commented on pages/dkp/konvoy/2.2/release-notes/2.2.1/index.md:
```diff
@@ -0,0 +1,105 @@
+| ----------------------- | ---------------------------------------------- | ------- | ----------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Cert Manager            | cert-manager                                   | 1.7.1   | - chart: 1.7.1 `<br>`- cert-manager: 1.7.1                                                                                                          |
+| Chartmuseum             | chartmuseum                                    | 3.6.2   | - chart: 3.6.2 `<br>`- chartmuseum: 3.6.2                                                                                                           |
+| Containerd              | containerd                                     | 1.4.13  |                                                                                                                                                       |
```
```suggestion
| Containerd              | containerd                                     | 1.4.11  |                                                                                                                                                       |
```

Colin1841 commented:
@cneth @faiq - updates & changes applied, and added a new "Other changes and capabilities" section for the flatcar note.
Please verify!
Ignore the 2.2.0 file, I just noticed a few spacing issues there that I cleaned up.

Colin1841 commented:
@mikolajb - Added the install known issue to this PR - see Known Issues section.
@cneth @msdolbey - please take a look at the added Known issue from https://jira.d2iq.com/browse/D2IQ-89688

dkoshkin commented on pages/dkp/konvoy/2.2/release-notes/2.2.1/index.md:
```diff
@@ -0,0 +1,123 @@
+| ------------------ | ------- |
+| **Minimum**  | 1.21.0  |
+| **Maximum**  | 1.22.x  |
+| **Default**  | 1.22.0  |
```
```suggestion
| **Default**  | 1.22.8  |
```

faiq commented on pages/dkp/konvoy/2.2/release-notes/2.2.1/index.md:
```diff
@@ -0,0 +1,123 @@
+| ------------------ | ------- |
+| **Minimum**  | 1.21.0  |
+| **Maximum**  | 1.22.x  |
+| **Default**  | 1.22.0  |
```
```suggestion
| **Default**  | 1.22.8  |
```

I believe the default version is 1.22.8

faiq replied:
See code: https://github.com/mesosphere/konvoy2/blob/v2.2.1-rc.3/pkg/api/v1alpha1/types.go#L15

cneth commented on pages/dkp/kommander/2.2/release-notes/2.2.1/index.md:
```diff
@@ -0,0 +1,123 @@
+
+The following items are known issues with this release.
+
+### Unsupported installation parameters
+
+If you set any these parameters in `kommander-vars` ConfigMap, the upgrade process fails.
+
+-   appManagementImageTag
+-   appManagementImageRepository
+-   kommanderChartsVersion
+
+Setting kommander chart version during installation configuration results in setting above values in kommander-vars config map, which then can block the upgrade. To unblock the upgrade, you can edit and remove the Config map values.
+Starting with release 2.2.0, you can specify the kommander chart version during installation configuration. However, if you manually change the value, then you might not be able to upgrade to higher versions.
+
+We recommend you DO NOT change the default value of the kommander chart version during installation. If you change this value, the outcome of such an installation is unpredictable. kommander-vars can be edited but it will cause an immediate restart to kommander 2.2.0. Without removing these vars, kommander cannot be upgraded.
+<QUESTION: do we want to recommend to contact support if this situation occurs and the user is blocked from upgrading?>
```
@mikolajb 

We need to rework this a bit, as it's not at all clear how a customer gets to this point (I don't think deliberately changing the chart version is a likely scenario.   But we do tell people to set the appManagementImageRepository in airgapped deployments, so we need to understand what, if anything that effects.
 
I put some questions in the JIRA (https://jira.d2iq.com/browse/D2IQ-89688), i think it will be clearer after we get those what we need to say here.

mmcelroy91 commented on pages/dkp/kommander/2.2/release-notes/2.2.1/index.md:
```diff
@@ -0,0 +1,123 @@
+<!-- Add links to external documentation as needed -->
+
+For more information about working with native Kubernetes, see the [Kubernetes documentation][kubernetes-doc].
+
```
```suggestion
For a full list of attributed 3rd party software, see [d2iq.com/legal/3rd](http://d2iq.com/legal/3rd).
```
Since this has the purple download Kommander button on it, do we need the legal attribution for MinIO here at the bottom?

mmcelroy91 commented on pages/dkp/konvoy/2.2/release-notes/2.2.1/index.md:
```diff
@@ -0,0 +1,123 @@
+<!-- Add links to external documentation as needed -->
+
+For more information about working with native Kubernetes, see the [Kubernetes documentation][kubernetes-doc].
+
```
```suggestion
For a full list of attributed 3rd party software, see [d2iq.com/legal/3rd](http://d2iq.com/legal/3rd)
```

mmcelroy91 replied:
Same questions as above.

cmastr reviewed (changes requested):
From Dimitri:

All the names are the same, just replace v2.2.0 with v2.2.1 for any artifact that is generated by us.

cneth commented on pages/dkp/kommander/2.2/release-notes/2.2.1/index.md:
```diff
@@ -0,0 +1,123 @@
+
+The following items are known issues with this release.
+
+### Unsupported installation parameters
+
+If you set any these parameters in `kommander-vars` ConfigMap, the upgrade process fails.
+
+-   appManagementImageTag
+-   appManagementImageRepository
+-   kommanderChartsVersion
+
+Setting kommander chart version during installation configuration results in setting above values in kommander-vars config map, which then can block the upgrade. To unblock the upgrade, you can edit and remove the Config map values.
+Starting with release 2.2.0, you can specify the kommander chart version during installation configuration. However, if you manually change the value, then you might not be able to upgrade to higher versions.
+
+We recommend you DO NOT change the default value of the kommander chart version during installation. If you change this value, the outcome of such an installation is unpredictable. kommander-vars can be edited but it will cause an immediate restart to kommander 2.2.0. Without removing these vars, kommander cannot be upgraded.
+<QUESTION: do we want to recommend to contact support if this situation occurs and the user is blocked from upgrading?>
```
I think maybe Mikolaj is out for the day, so I'd like to suggest we add this for now, and if necessary, we can iterate on this going forwards.

```suggestion
### Upgrade considerations

Before attempting to upgrade an existing cluster to this release, please check the 'kommander-vars' Configmap in the 'kommander' namespace for the following fields:

-   appManagementImageTag
-   appManagementImageRepository
-   kommanderChartsVersion

If any of the these fields are present, then there is a possibility the upgrade can fail.  If you encounter this situation, please file a support ticket for advice on how to remediate the issue before attempting to continue the upgrade.
```

mikolajb replied:
Sounds good to me. All variable names have prefix `kommander` in 2.2 (there was a mistake in a ticket description).
so the list is:
- `kommanderAppManagementImageTag`
- `kommanderAppManagementImageRepository`
- `kommanderChartsVersion`

mikolajb commented on pages/dkp/kommander/2.2/release-notes/2.2.1/index.md:
```diff
@@ -0,0 +1,123 @@
+
+If you set any these parameters in `kommander-vars` ConfigMap, the upgrade process fails.
+
+-   appManagementImageTag
+-   appManagementImageRepository
```
```suggestion
-   kommanderAppManagementImageTag
-   kommanderAppManagementImageRepository
```
Sorry, in the ticket description it wasn't correct.

cneth commented on pages/dkp/konvoy/2.2/release-notes/2.2.0/index.md:
```diff
@@ -353,6 +352,20 @@ Upgrading catalog applications using Spark Operator can fail when running `dkp u
+
+If you set any these parameters in `kommander-vars` ConfigMap, the upgrade process fails.
+
+-   appManagementImageTag
+-   appManagementImageRepository
```
```suggestion
-   kommanderAppManagementImageTag
-   kommanderAppManagementImageRepository
```

cneth commented on pages/dkp/konvoy/2.2/release-notes/2.2.0/index.md:
````diff
@@ -353,6 +352,20 @@ Upgrading catalog applications using Spark Operator can fail when running `dkp u
     kubectl delete pod -n $WORKSPACE_NAMESPACE $(kubectl get pod -l app.kubernetes.io/name=$SPARK_OPERATOR_RELEASE_NAME -n $WORKSPACE_NAMESPACE -o jsonpath='{range .items[0]}{.metadata.name}')
     ```
 
+### Unsupported installation parameters
+
+If you set any these parameters in `kommander-vars` ConfigMap, the upgrade process fails.
+
+-   kommanderAppManagementImageTag
+-   kommanderAppManagementImageRepository
+-   kommanderChartsVersion
+
+Setting kommander chart version during installation configuration results in setting above values in kommander-vars config map, which then can block the upgrade. To unblock the upgrade, you can edit and remove the Config map values.
+Starting with release 2.2.0, you can specify the kommander chart version during installation configuration. However, if you manually change the value, then you might not be able to upgrade to higher versions.
+
+We recommend you DO NOT change the default value of the kommander chart version during installation. If you change this value, the outcome of such an installation is unpredictable. kommander-vars can be edited but it will cause an immediate restart to kommander 2.2.0. Without removing these vars, kommander cannot be upgraded.
+<QUESTION: do we want to recommend to contact support if this situation occurs and the user is blocked from upgrading?>
+
````
This issue cannot happen on an upgrade to 2.2.0, so we don't need this here.

```suggestion
```

cneth commented on pages/dkp/konvoy/2.2/release-notes/2.2.1/index.md:
```diff
@@ -0,0 +1,123 @@
+
+The following items are known issues with this release.
+
+### Unsupported installation parameters
+
+If you set any these parameters in `kommander-vars` ConfigMap, the upgrade process fails.
+
+-   appManagementImageTag
+-   appManagementImageRepository
+-   kommanderChartsVersion
+
+Setting kommander chart version during installation configuration results in setting above values in kommander-vars config map, which then can block the upgrade. To unblock the upgrade, you can edit and remove the Config map values.
+Starting with release 2.2.0, you can specify the kommander chart version during installation configuration. However, if you manually change the value, then you might not be able to upgrade to higher versions.
+
+We recommend you DO NOT change the default value of the kommander chart version during installation. If you change this value, the outcome of such an installation is unpredictable. kommander-vars can be edited but it will cause an immediate restart to kommander 2.2.0. Without removing these vars, kommander cannot be upgraded.
+<QUESTION: do we want to recommend to contact support if this situation occurs and the user is blocked from upgrading?>
```
```suggestion
### Upgrade considerations

Before attempting to upgrade an existing cluster to this release, please check the 'kommander-vars' Configmap in the 'kommander' namespace for the following fields:

-   kommanderAppManagementImageTag
-   kommanderAppManagementImageRepository
-   kommanderChartsVersion

If any of the these fields are present, then there is a possibility the upgrade can fail.  If you encounter this situation, please file a support ticket for advice on how to remediate the issue before attempting to continue the upgrade.
```

Colin1841 replied:
```suggestion
### Unsupported installation parameters

Before attempting to upgrade an existing cluster to this release, please check the 'kommander-vars' Configmap in the 'kommander' namespace for the following fields:

-   kommanderAppManagementImageTag
-   kommanderAppManagementImageRepository
-   kommanderChartsVersion

If any of the these fields are present, then there is a possibility the upgrade can fail.  If you encounter this situation, please file a support ticket for advice on how to remediate the issue before attempting to continue the upgrade.
```

cmastr commented on pages/dkp/konvoy/2.2/release-notes/2.2.1/index.md:
```diff
@@ -0,0 +1,119 @@
+
+### Upgrade considerations
+
+Before attempting to upgrade an existing cluster to this release, please check the 'kommander-vars' Configmap in the 'kommander' namespace for the following fields:
```
```suggestion
Before attempting to upgrade an existing cluster to this release, check the 'kommander-vars' Configmap in the 'kommander' namespace for the following fields:
```

cmastr commented on pages/dkp/konvoy/2.2/release-notes/2.2.1/index.md:
```diff
@@ -0,0 +1,119 @@
+-   kommanderAppManagementImageRepository
+-   kommanderChartsVersion
+
+If any of the these fields are present, then there is a possibility the upgrade can fail.  If you encounter this situation, please file a support ticket for advice on how to remediate the issue before attempting to continue the upgrade.
```
```suggestion
If any of the these fields are present, then there is a possibility the upgrade can fail.  If you encounter this situation, file a support ticket for advice on how to remediate the issue before attempting to continue the upgrade.
```

Colin1841 replied:
Argh, darn those please usages!

Colin1841 commented on pages/dkp/kommander/2.2/release-notes/2.2.1/index.md:
```diff
@@ -0,0 +1,119 @@
+enterprise: false
+beta: false
+---
+**D2iQ&reg; Konvoy&reg; version 2.2.1 was released on May 26, 2022.**
```
```suggestion
**D2iQ&reg; Konvoy&reg; version 2.2.1 was released on May 31, 2022.**
```

Colin1841 commented on pages/dkp/konvoy/2.2/release-notes/2.2.1/index.md:
```diff
@@ -0,0 +1,119 @@
+enterprise: false
+beta: false
+---
+**D2iQ&reg; Konvoy&reg; version 2.2.1 was released on May 26, 2022.**
```
```suggestion
**D2iQ&reg; Konvoy&reg; version 2.2.1 was released on May 31, 2022.**
```
